install-steps.md

```diff
@@ -1,193 +1,850 @@
+3. Configure VARS
+    ~~~sh
+    export VERSION="4.3.0-0.nightly-2019-12-09-035405"
+    export RELEASE_IMAGE=$(curl -s https://mirror.openshift.com/pub/openshift-v4/clients/ocp-dev-preview/$VERSION/release.txt | grep 'Pull From: quay.io' | awk -F ' ' '{print $3}' | xargs)
+    ~~~
+
+### Choosing the latest OpenShift Installer
+
+1. Configure VARS
+    ~~~sh
+    export VERSION=$(curl -s https://mirror.openshift.com/pub/openshift-v4/clients/ocp-dev-preview/latest/release.txt | grep 'Name:' | awk -F: '{print $2}' | xargs)
+    export RELEASE_IMAGE=$(curl -s https://mirror.openshift.com/pub/openshift-v4/clients/ocp-dev-preview/latest/release.txt | grep 'Pull From: quay.io' | awk -F ' ' '{print $3}' | xargs)
+    ~~~
+
+### Extract the Installer
+
+Once the installer has been chosen, the next step is to extract it. 
+
+~~~sh
 export cmd=openshift-baremetal-install
-export pullsecret_file=pull-secret.txt
+export pullsecret_file=~/pull-secret.txt
 export extract_dir=$(pwd)
-curl -s https://mirror.openshift.com/pub/openshift-v4/clients/ocp-dev-preview/latest/openshift-client-linux-$VERSION.tar.gz | tar zxvf - oc
-sudo cp oc /usr/local/bin/
-```
+# Get the oc binary
+curl -s https://mirror.openshift.com/pub/openshift-v4/clients/ocp-dev-preview/$VERSION/openshift-client-linux-$VERSION.tar.gz | tar zxvf - oc
+# Extract the baremetal installer
+./oc adm release extract --registry-config "${pullsecret_file}" --command=$cmd --to "${extract_dir}" ${RELEASE_IMAGE}
+~~~
+
+## Configure the install-config and run the installer
+
+1. Configure the `install-config.yaml` (Make sure you change the pullSecret and sshKey)
+    ~~~yaml
+    apiVersion: v1
+    baseDomain: <domain>
+    metadata:
+      name: <cluster-name>
+    networking:
+      machineCIDR: <public-cidr>
+    compute:
+    - name: worker
+      replicas: 2
+    controlPlane:
+      name: master
+      replicas: 3
+      platform:
+        baremetal: {}
+    platform:
+      baremetal:
+        apiVIP: <api-ip>
+        ingressVIP: <wildcard-ip>
+        dnsVIP: <dns-ip>
+        provisioningBridge: provisioning
+        externalBridge: baremetal
+        hosts:
+          - name: openshift-master-0
+            role: master
+            bmc:
+              address: ipmi://<out-of-band-ip>
+              username: <user>
+              password: <password>
+            bootMACAddress: <NIC1-mac-address>
+            hardwareProfile: default
+          - name: openshift-master-1
+            role: master
+            bmc:
+              address: ipmi://<out-of-band-ip>
+              username: <user>
+              password: <password>
+            bootMACAddress: <NIC1-mac-address>
+            hardwareProfile: default
+          - name: openshift-master-2
+            role: master
+            bmc:
+              address: ipmi://<out-of-band-ip>
+              username: <user>
+              password: <password>
+            bootMACAddress: <NIC1-mac-address
+            hardwareProfile: unknown
```

Comment:
The hardwareProfile settings don't match throughout this example. I'm not sure if the installer uses different values ("default") from metal3 ("unknown"), so I don't know which to suggest you use. You could probably leave them out entirely except in cases where you need to be explicit to ensure the right disk is selected for the deployment.

Reply:
@dh

> The hardwareProfile settings don't match throughout this example. I'm not sure if the installer uses different values ("default") from metal3 ("unknown"), so I don't know which to suggest you use. You could probably leave them out entirely except in cases where you need to be explicit to ensure the right disk is selected for the deployment.

@dhellmann my understanding is the master nodes require default and the worker nodes require unknown from what @e-minguez told me. I also tested an env this way where masters were default and workers were unknown and it worked for me.

Reply:
OK. Master 0 and 1 are using "default" and master 2 is using "unknown". Worker 0 is using "unknown" and worker 1 is using "default".

Reply:
nice catch. fixed.